ThunkLibs/libvulkan/Host.cpp

```diff
@@ -172,6 +240,25 @@ static PFN_vkVoidFunction FEXFN_IMPL(vkGetInstanceProcAddr)(VkInstance a_0, cons
 
   // Check for functions with custom implementations first
   if (auto ptr = LookupCustomVulkanFunction(a_1)) {
+    // If this function belongs to an instance extension, requery its address.
+    // This ensures fexldr_ptr_* is valid if the application creates a minimal
+    // VkInstance with no extensions before creating its actual instance.
+    using namespace std::string_view_literals;
+    if (a_1 == "vkGetRandROutputDisplayEXT"sv && !LDR_PTR(vkGetRandROutputDisplayEXT)) {
+      (void*&)LDR_PTR(vkGetRandROutputDisplayEXT) = (void*)LDR_PTR(vkGetInstanceProcAddr)(a_0, "vkGetRandROutputDisplayEXT");
+    }
+    if (a_1 == "vkAcquireXlibDisplayEXT"sv && !LDR_PTR(vkAcquireXlibDisplayEXT)) {
+      (void*&)LDR_PTR(vkAcquireXlibDisplayEXT) = (void*)LDR_PTR(vkGetInstanceProcAddr)(a_0, "vkAcquireXlibDisplayEXT");
+    }
+    if (a_1 == "vkGetPhysicalDeviceXcbPresentationSupportKHR"sv && !LDR_PTR(vkGetPhysicalDeviceXcbPresentationSupportKHR)) {
+      (void*&)LDR_PTR(vkGetPhysicalDeviceXcbPresentationSupportKHR) = (void*)LDR_PTR(vkGetInstanceProcAddr)(a_0, "vkGetPhysicalDeviceXcbPre"
```

Comment:
This weird string break should be reformatted to not look terrible. Same with the one in the next if statement.

Reply:
I've been trying a couple of variations on that line and couldn't get `clang-format` to produce a different result. We could turn off formatting locally, but not sure the two-line preprocessor noise added by that is any better.

Happy to proceed with whichever route you prefer.

Reply:
Ah yes, hadn't tried a local variable. Definitely less disruptive than the options I considered :)